rust/src/request.rs

```diff
@@ -0,0 +1,189 @@
+use crate::{
+    client::ClientError,
+    crypto::ByteObject,
+    request::Proxy::{InMem, Remote},
+    service::{data::RoundParametersData, Handle},
+    ParticipantPublicKey,
+    SumDict,
+    UpdateSeedDict,
+};
+use bytes::Bytes;
+use reqwest::{Client, Error, IntoUrl, Response, StatusCode};
+
+/// Proxy for communicating with the service.
+pub enum Proxy {
```

Comment:
i like this suggestion! the only (perhaps minor) drawbacks i can think of are

- if we want to add or change operations in the future (and I expect it's likely - I had to refactor `get_sum_dict` a couple times alone doing the rebase), it's not totally painless: we'd need to change in multiple places: the trait, and also the `impl`s. with the existing enum based `Proxy`, it's just 1 change e.g. add 1 new function `get_scalar`.
- it requires a dependency, due to it being async

I didn't apply your suggestion yet because there are enough changes already (and the enum-based `Proxy` works well enough for now), but i think it's an optimisation we can definitely consider in the next update!